### PR TITLE
APEX-16: Code style checks- only for review

### PR DESCRIPTION
**Pull request by chandnisingh**

### CODE REVIEW

File: dt_checks.xml [hunk 1]
```diff
@@ -10,4 +10,110 @@
...
+
+    <module name="RightCurly">
+      <property name="option" value="same"/>
+      <property name="tokens" value="LITERAL_TRY,LITERAL_CATCH,LITERAL_FINALLY,LITERAL_IF,LITERAL_ELSE"/>
+    </module>
+
+    <module name="RightCurly">
+      <property name="option" value="alone"/>
+      <property name="tokens"
+                value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
```
Comment (243826) [inline dt_checks.xml]: is there a difference when you put a space after `,` and no space after `,`?

File: dt_checks.xml [hunk 2]
```diff
@@ -10,4 +10,110 @@
...
+    </module>
+
+    <module name="NoFinalizer"/>
+    <module name="GenericWhitespace">
+      <message key="ws.followed"
+               value="GenericWhitespace ''{0}'' is followed by whitespace."/>
+      <message key="ws.preceded"
+               value="GenericWhitespace ''{0}'' is preceded with whitespace."/>
+      <message key="ws.illegalFollow"
+               value="GenericWhitespace ''{0}'' should followed by whitespace."/>
```
Comment (243826) [inline dt_checks.xml]: `be ` missing?

**Comment by 243826** [comment]
overall it looks good to me. Can you enable it such that only if more deviations are added, the build will fail? And make it chatty only when a system property is specified while building.

File: dt_checks.xml (hunk 1 above)
Comment (chandnisingh) [inline dt_checks.xml]: No will fix that

File: dt_checks.xml (hunk 2 above)
Comment (chandnisingh) [inline dt_checks.xml]: Will fix it

**Comment by chandnisingh** [comment]
Will count existing deviations and will add a check that if more are introduced then build will fail. 
Also should we add a check for license?

**Comment by 243826** [comment]
check for license is already there (mvn license:check i think). And I think
that will be more comprehensive. But if Checkstyle does at least as good of
a job, I think it may warrant consolidation.

**Pull request closed without merging**